lib/cfa/augeas_parser.rb

```diff
@@ -112,6 +122,11 @@ def delete(key)
       @data.reject! { |entry| entry[:key] == key }
     end
 
+    # @yieldparam block [Object] condition to satisfy
+    def delete_if(&_block)
```

Comment:
maybe also new method is not needed and passing matcher to delete itself, can make sense